python_modules/jupedsim/jupedsim/sqlite_serialization.py

```diff
@@ -44,20 +45,23 @@ def __init__(
                 Note: the file will not be written until the first call to :func:`begin_writing`
             every_nth_frame: int
                 indicates interval between writes, 1 means every frame, 5 every 5th
-
-        Returns:
-            SqliteTrajectoryWriter
+            commit_every_nth_write: int
+                number of frames to keep in RAM before writing them to disk.
         """
         self._output_file = output_file
         if every_nth_frame < 1:
             raise TrajectoryWriter.Exception("'every_nth_frame' has to be > 0")
         self._every_nth_frame = every_nth_frame
-        self._con = sqlite3.connect(self._output_file, isolation_level=None)
+        self._con = sqlite3.connect(self._output_file)
         # Don't wait for the OS to persist data
         self._con.execute("PRAGMA synchronous=OFF;")
         # Don't allow rollbacks (we don't have need for it)
         self._con.execute("PRAGMA journal_mode=OFF;")
 
+        # Buffering options
+        self._commit_every_nth_write = int(commit_every_nth_write) if commit_every_nth_write > 0 else 1
```

Comment:
The general philosophy in JuPedSim is that we try not to anticipate what users want or try to fix user input errors. This needs to throw an exception if `commit_every_nth_write` is less that 1